exchanges/kucoin/kucoin_types.go

```diff
@@ -65,20 +67,12 @@ func (k kucoinTimeMilliSec) Time() time.Time {
 	return time.UnixMilli(int64(k))
 }
 
-// kucoinTimeMilliSecStr provides an internal conversion helper
-type kucoinTimeMilliSecStr time.Time
-
-// Time returns a time.Time object
-func (k kucoinTimeMilliSecStr) Time() time.Time {
-	return time.Time(k)
-}
-
 // kucoinTimeNanoSec provides an internal conversion helper
-type kucoinTimeNanoSec time.Time
+type kucoinTimeNanoSec int64
 
 // Time returns a time.Time object
-func (k kucoinTimeNanoSec) Time() time.Time {
-	return time.Time(k)
+func (k *kucoinTimeNanoSec) Time() time.Time {
+	return time.Unix(0, int64(*k))
```

Comment:
NOTE: A zero value using unix function creates a non-zero time. So I might suggest we use an integer < 0 to then return an empty `time.Time{}` value.